specifyweb/stored_queries/query_construct.py

```diff
@@ -15,10 +15,12 @@ def get_treedef(collection, tree_name):
             if tree_name == 'Storage' else
             getattr(collection.discipline, tree_name.lower() + "treedef"))
 
-class QueryConstruct(namedtuple('QueryConstruct', 'collection objectformatter query join_cache param_count')):
+class QueryConstruct(namedtuple('QueryConstruct', 'collection objectformatter query join_cache param_count tree_rank_count')):
+
     def __new__(cls, *args, **kwargs):
         kwargs['join_cache'] = dict()
         kwargs['param_count'] = 0
+        kwargs['tree_rank_count'] = 0
```

Comment:
Outside of these two usages, where is `QueryConstruct.tree_rank_count` used? I can not find another use in the code base for the attribute.

Reply:
Yeah, @acwhite211 you can rid of this. I was working on handling cases where formatter of taxon is defined with fields from the parent. In that case, the cycle will end (unlike other cyclical cases). Alec you can continue on that if you want!

Reply:
Sounds good, I'll keep it in, I added a TODO note in the code on implementing this in the future.